CONJUNTO DE DATOS

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('/workspaces/Proyecto_nocountry/data/raw/HDHI Admission data.csv')
df

SNO MRD No.       D.O.A       D.O.D  AGE GENDER RURAL  \
0          1  234735    4/1/2017    4/3/2017   81      M     R   
1          2  234696    4/1/2017    4/5/2017   65      M     R   
2          3  234882    4/1/2017    4/3/2017   53      M     U   
3          4  234635    4/1/2017    4/8/2017   67      F     U   
4          5  234486    4/1/2017   4/23/2017   60      F     U   
...      ...     ...         ...         ...  ...    ...   ...   
15752  15753  699585  31/03/2019  04/04/2019   86      F     U   
15753  15754  699500   3/31/2019    4/1/2019   50      M     R   
15754  15755  700415  31/03/2019  09/04/2019   82      M     U   
15755  15756  699524  31/03/2019  03/04/2019   59      F     U   
15756  15757  699524  31/03/2019  03/04/2019   59      F     U   

      TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  ...  \
0                                   E     Apr-17                 3  ...   
1                                   E     Apr-17                 5  ...   
2                                   E     Apr-17                 3  ...   
3                                   E     Apr-17                 8  ...   
4                                   E     Apr-17                23  ...   
...                               ...        ...               ...  ...   
15752                               O     Mar-19                 5  ...   
15753                               E     Mar-19                 2  ...   
15754                               E     Mar-19                10  ...   
15755                               O     Mar-19                 4  ...   
15756                               O     Mar-19                 4  ...   

       CONGENITAL UTI  NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  \
0               0   0                          0            0   
1               0   0                          0            0   
2               0   0                          0            0   
3               0   0                          0            0   
4               0   0                          0            0   
...           ...  ..                        ...          ...   
15752           0   0                          0            0   
15753           0   0                          0            0   
15754           0   0                          0            0   
15755           0   0                          0            0   
15756           0   0                          0            0   

       INFECTIVE ENDOCARDITIS  DVT  CARDIOGENIC SHOCK  SHOCK  \
0                           0    0                  0      0   
1                           0    0                  0      0   
2                           0    0                  0      0   
3                           0    0                  0      0   
4                           0    0                  0      0   
...                       ...  ...                ...    ...   
15752                       0    0                  0      0   
15753                       0    0                  0      0   
15754                       0    0                  0      0   
15755                       0    0                  0      0   
15756                       0    0                  0      0   

       PULMONARY EMBOLISM CHEST INFECTION  
0                       0               0  
1                       0               0  
2                       0               0  
3                       0               0  
4                       0               0  
...                   ...             ...  
15752                   0               0  
15753                   0               0  
15754                   0               0  
15755                   0               0  
15756                   0               0  

[15757 rows x 56 columns]

In [4]:
#Se renombra columna
df = df.rename(columns={'MRD No.': 'ID'})

In [5]:
#Se renombra columna
df = df.rename(columns={'SNO': 'ADMISION'})
df

ADMISION      ID       D.O.A       D.O.D  AGE GENDER RURAL  \
0             1  234735    4/1/2017    4/3/2017   81      M     R   
1             2  234696    4/1/2017    4/5/2017   65      M     R   
2             3  234882    4/1/2017    4/3/2017   53      M     U   
3             4  234635    4/1/2017    4/8/2017   67      F     U   
4             5  234486    4/1/2017   4/23/2017   60      F     U   
...         ...     ...         ...         ...  ...    ...   ...   
15752     15753  699585  31/03/2019  04/04/2019   86      F     U   
15753     15754  699500   3/31/2019    4/1/2019   50      M     R   
15754     15755  700415  31/03/2019  09/04/2019   82      M     U   
15755     15756  699524  31/03/2019  03/04/2019   59      F     U   
15756     15757  699524  31/03/2019  03/04/2019   59      F     U   

      TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  ...  \
0                                   E     Apr-17                 3  ...   
1                                   E     Apr-17                 5  ...   
2                                   E     Apr-17                 3  ...   
3                                   E     Apr-17                 8  ...   
4                                   E     Apr-17                23  ...   
...                               ...        ...               ...  ...   
15752                               O     Mar-19                 5  ...   
15753                               E     Mar-19                 2  ...   
15754                               E     Mar-19                10  ...   
15755                               O     Mar-19                 4  ...   
15756                               O     Mar-19                 4  ...   

       CONGENITAL UTI  NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  \
0               0   0                          0            0   
1               0   0                          0            0   
2               0   0                          0            0   
3               0   0                          0            0   
4               0   0                          0            0   
...           ...  ..                        ...          ...   
15752           0   0                          0            0   
15753           0   0                          0            0   
15754           0   0                          0            0   
15755           0   0                          0            0   
15756           0   0                          0            0   

       INFECTIVE ENDOCARDITIS  DVT  CARDIOGENIC SHOCK  SHOCK  \
0                           0    0                  0      0   
1                           0    0                  0      0   
2                           0    0                  0      0   
3                           0    0                  0      0   
4                           0    0                  0      0   
...                       ...  ...                ...    ...   
15752                       0    0                  0      0   
15753                       0    0                  0      0   
15754                       0    0                  0      0   
15755                       0    0                  0      0   
15756                       0    0                  0      0   

       PULMONARY EMBOLISM CHEST INFECTION  
0                       0               0  
1                       0               0  
2                       0               0  
3                       0               0  
4                       0               0  
...                   ...             ...  
15752                   0               0  
15753                   0               0  
15754                   0               0  
15755                   0               0  
15756                   0               0  

[15757 rows x 56 columns]

In [6]:
#Se renombra indice como ID
df.index = df['ADMISION']
df

ADMISION      ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                               
1                1  234735    4/1/2017    4/3/2017   81      M     R   
2                2  234696    4/1/2017    4/5/2017   65      M     R   
3                3  234882    4/1/2017    4/3/2017   53      M     U   
4                4  234635    4/1/2017    4/8/2017   67      F     U   
5                5  234486    4/1/2017   4/23/2017   60      F     U   
...            ...     ...         ...         ...  ...    ...   ...   
15753        15753  699585  31/03/2019  04/04/2019   86      F     U   
15754        15754  699500   3/31/2019    4/1/2019   50      M     R   
15755        15755  700415  31/03/2019  09/04/2019   82      M     U   
15756        15756  699524  31/03/2019  03/04/2019   59      F     U   
15757        15757  699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  ...  \
ADMISION                                                               ...   
1                                      E     Apr-17                 3  ...   
2                                      E     Apr-17                 5  ...   
3                                      E     Apr-17                 3  ...   
4                                      E     Apr-17                 8  ...   
5                                      E     Apr-17                23  ...   
...                                  ...        ...               ...  ...   
15753                                  O     Mar-19                 5  ...   
15754                                  E     Mar-19                 2  ...   
15755                                  E     Mar-19                10  ...   
15756                                  O     Mar-19                 4  ...   
15757                                  O     Mar-19                 4  ...   

          CONGENITAL UTI  NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  \
ADMISION                                                           
1                  0   0                          0            0   
2                  0   0                          0            0   
3                  0   0                          0            0   
4                  0   0                          0            0   
5                  0   0                          0            0   
...              ...  ..                        ...          ...   
15753              0   0                          0            0   
15754              0   0                          0            0   
15755              0   0                          0            0   
15756              0   0                          0            0   
15757              0   0                          0            0   

          INFECTIVE ENDOCARDITIS  DVT  CARDIOGENIC SHOCK  SHOCK  \
ADMISION                                                          
1                              0    0                  0      0   
2                              0    0                  0      0   
3                              0    0                  0      0   
4                              0    0                  0      0   
5                              0    0                  0      0   
...                          ...  ...                ...    ...   
15753                          0    0                  0      0   
15754                          0    0                  0      0   
15755                          0    0                  0      0   
15756                          0    0                  0      0   
15757                          0    0                  0      0   

          PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                      
1                          0               0  
2                          0               0  
3                          0               0  
4                          0               0  
5                  

In [7]:
#Elimino columna duplicada ID
del df[df.columns[0]]

In [8]:
df

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
1         234735    4/1/2017    4/3/2017   81      M     R   
2         234696    4/1/2017    4/5/2017   65      M     R   
3         234882    4/1/2017    4/3/2017   53      M     U   
4         234635    4/1/2017    4/8/2017   67      F     U   
5         234486    4/1/2017   4/23/2017   60      F     U   
...          ...         ...         ...  ...    ...   ...   
15753     699585  31/03/2019  04/04/2019   86      F     U   
15754     699500   3/31/2019    4/1/2019   50      M     R   
15755     700415  31/03/2019  09/04/2019   82      M     U   
15756     699524  31/03/2019  03/04/2019   59      F     U   
15757     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
1                                      E     Apr-17                 3   
2                                      E     Apr-17                 5   
3                                      E     Apr-17                 3   
4                                      E     Apr-17                 8   
5                                      E     Apr-17                23   
...                                  ...        ...               ...   
15753                                  O     Mar-19                 5   
15754                                  E     Mar-19                 2   
15755                                  E     Mar-19                10   
15756                                  O     Mar-19                 4   
15757                                  O     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
1                                       2  ...          0    0   
2                                       2  ...          0    0   
3                                       3  ...          0    0   
4                                       6  ...          0    0   
5                                       9  ...          0    0   
...                                   ...  ...        ...  ...   
15753                                   1  ...          0    0   
15754                                   2  ...          0    0   
15755                                   5  ...          0    0   
15756                                   2  ...          0    0   
15757                                   2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
1                                 0            0                       0    0   
2                                 0            0                       0    0   
3                                 0            0                       0    0   
4                                 0            0                       0    0   
5                                 0            0                       0    0   
...                             ...          ...                     ...  ...   
15753                             0            0                       0    0   
15754                             0            0                       0    0   
15755                             0            0                       0    0   
15756                             0            0                       0    0   
15757                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
1                         0      0                  0               0  
2                         0      0                  0               0  
3                         0      0                  0   

EXPLORACIÓN Y LIMPIEZA DE DATOS

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 15757 entries, 1 to 15757
Data columns (total 55 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   ID                               15757 non-null  object
 1   D.O.A                            15757 non-null  object
 2   D.O.D                            15757 non-null  object
 3   AGE                              15757 non-null  int64 
 4   GENDER                           15757 non-null  object
 5   RURAL                            15757 non-null  object
 6   TYPE OF ADMISSION-EMERGENCY/OPD  15757 non-null  object
 7   month year                       15757 non-null  object
 8   DURATION OF STAY                 15757 non-null  int64 
 9   duration of intensive unit stay  15757 non-null  int64 
 10  OUTCOME                          15757 non-null  object
 11  SMOKING                          15757 non-null  int64 
 12  ALCOHOL                          1575

In [10]:
#Mostrando un listado de todas las columnas
list(df.columns)

['ID',
 'D.O.A',
 'D.O.D',
 'AGE',
 'GENDER',
 'RURAL',
 'TYPE OF ADMISSION-EMERGENCY/OPD',
 'month year',
 'DURATION OF STAY',
 'duration of intensive unit stay',
 'OUTCOME',
 'SMOKING ',
 'ALCOHOL',
 'DM',
 'HTN',
 'CAD',
 'PRIOR CMP',
 'CKD',
 'HB',
 'TLC',
 'PLATELETS',
 'GLUCOSE',
 'UREA',
 'CREATININE',
 'BNP',
 'RAISED CARDIAC ENZYMES',
 'EF',
 'SEVERE ANAEMIA',
 'ANAEMIA',
 'STABLE ANGINA',
 'ACS',
 'STEMI',
 'ATYPICAL CHEST PAIN',
 'HEART FAILURE',
 'HFREF',
 'HFNEF',
 'VALVULAR',
 'CHB',
 'SSS',
 'AKI',
 'CVA INFRACT',
 'CVA BLEED',
 'AF',
 'VT',
 'PSVT',
 'CONGENITAL',
 'UTI',
 'NEURO CARDIOGENIC SYNCOPE',
 'ORTHOSTATIC',
 'INFECTIVE ENDOCARDITIS',
 'DVT',
 'CARDIOGENIC SHOCK',
 'SHOCK',
 'PULMONARY EMBOLISM',
 'CHEST INFECTION']

In [11]:
#Sumar la cantidad de nulos
df.isnull().sum()[df.isnull().sum()>0]

HB             252
TLC            286
PLATELETS      285
GLUCOSE        863
UREA           241
CREATININE     247
BNP           8441
EF            1505
dtype: int64

In [12]:
#Buscar cuantos ID hay repetidos
df[df.duplicated('ID')]

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
17        413903  01/04/2017  03/04/2017   54      M     U   
91        382159    4/4/2017    4/9/2017   47      F     U   
96         48098    4/5/2017   4/10/2017   65      F     R   
131       384232  08/04/2017  12/04/2017   80      F     U   
148       203622  10/04/2017  17/04/2017   58      F     U   
...          ...         ...         ...  ...    ...   ...   
15744     699413  30/03/2019  02/04/2019   75      F     U   
15746     699480  30/03/2019  03/04/2019   60      F     U   
15749     273016  30/03/2019  31/03/2019   74      F     U   
15752     469963  31/03/2019  08/04/2019   60      F     U   
15757     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
17                                     O     Apr-17                 3   
91                                     E     Apr-17                 6   
96                                     O     Apr-17                 6   
131                                    E     Apr-17                 5   
148                                    O     Apr-17                 8   
...                                  ...        ...               ...   
15744                                  E     Mar-19                 4   
15746                                  E     Mar-19                 5   
15749                                  E     Mar-19                 2   
15752                                  E     Mar-19                 9   
15757                                  O     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
17                                      1  ...          0    0   
91                                      1  ...          0    0   
96                                      4  ...          0    0   
131                                     5  ...          0    0   
148                                     0  ...          0    1   
...                                   ...  ...        ...  ...   
15744                                   3  ...          0    0   
15746                                   4  ...          0    0   
15749                                   2  ...          0    0   
15752                                   9  ...          0    0   
15757                                   2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
17                                0            0                       0    0   
91                                0            0                       0    0   
96                                0            0                       0    0   
131                               0            0                       0    0   
148                               0            0                       0    0   
...                             ...          ...                     ...  ...   
15744                             0            0                       0    0   
15746                             0            0                       0    0   
15749                             0            0                       0    0   
15752                             0            0                       0    0   
15757                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
17                        0      0                  0               0  
91                        0      0                  0               0  
96                        1      1                  0   

In [13]:
df[df['ID'] =='95092']

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                    
14631     95092  15/02/2019  20/02/2019   77      M     U   
14632     95092  15/02/2019  20/02/2019   77      M     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
14631                                  E     Feb-19                 6   
14632                                  E     Feb-19                 6   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
14631                                   5  ...          0    0   
14632                                   5  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
14631                             0            0                       0    0   
14632                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
14631                     0      0                  0               0  
14632                     0      0                  0               0  

[2 rows x 55 columns]

In [14]:
df[df['ID'] == '699524']

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
15756     699524  31/03/2019  03/04/2019   59      F     U   
15757     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
15756                                  O     Mar-19                 4   
15757                                  O     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
15756                                   2  ...          0    0   
15757                                   2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
15756                             0            0                       0    0   
15757                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
15756                     0      0                  0               0  
15757                     0      0                  0               0  

[2 rows x 55 columns]

In [15]:
#Buscando duplicados en columnas de ID y DOA
duplicados = df[df.duplicated(subset=['ID', 'D.O.A'], keep=False)]
duplicados

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
303       250680   4/24/2017   4/27/2017   85      F     R   
321       250680   4/24/2017   4/27/2017   85      F     U   
380       235427   4/26/2017   4/28/2017   56      M     R   
383       235427   4/26/2017   4/28/2017   56      M     U   
452       253963   4/29/2017    5/2/2017   65      M     R   
...          ...         ...         ...  ...    ...   ...   
15744     699413  30/03/2019  02/04/2019   75      F     U   
15745     699480  30/03/2019  03/04/2019   60      F     U   
15746     699480  30/03/2019  03/04/2019   60      F     U   
15756     699524  31/03/2019  03/04/2019   59      F     U   
15757     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
303                                    E     Apr-17                 4   
321                                    E     Apr-17                 4   
380                                    O     Apr-17                 3   
383                                    O     Apr-17                 3   
452                                    E     Apr-17                 4   
...                                  ...        ...               ...   
15744                                  E     Mar-19                 4   
15745                                  E     Mar-19                 5   
15746                                  E     Mar-19                 5   
15756                                  O     Mar-19                 4   
15757                                  O     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
303                                     3  ...          0    0   
321                                     0  ...          0    0   
380                                     1  ...          0    0   
383                                     0  ...          0    0   
452                                     3  ...          0    0   
...                                   ...  ...        ...  ...   
15744                                   3  ...          0    0   
15745                                   4  ...          0    0   
15746                                   4  ...          0    0   
15756                                   2  ...          0    0   
15757                                   2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
303                               0            0                       0    0   
321                               0            0                       0    0   
380                               0            0                       0    0   
383                               0            0                       0    0   
452                               0            0                       0    0   
...                             ...          ...                     ...  ...   
15744                             0            0                       0    0   
15745                             0            0                       0    0   
15746                             0            0                       0    0   
15756                             0            0                       0    0   
15757                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
303                       0      0                  0               0  
321                       0      0                  0               0  
380                       0      0                  0   

In [16]:
df[df['ID']== '699413']

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
15743     699413  30/03/2019  02/04/2019   75      F     U   
15744     699413  30/03/2019  02/04/2019   75      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
15743                                  E     Mar-19                 4   
15744                                  E     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
15743                                   3  ...          0    0   
15744                                   3  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
15743                             0            0                       0    0   
15744                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
15743                     0      0                  0               0  
15744                     0      0                  0               0  

[2 rows x 55 columns]

In [17]:
df[df['ID']== '253963']

ID      D.O.A     D.O.D  AGE GENDER RURAL  \
ADMISION                                                  
452       253963  4/29/2017  5/2/2017   65      M     R   
462       253963  4/29/2017  5/2/2017   65      M     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
452                                    E     Apr-17                 4   
462                                    E     Apr-17                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
452                                     3  ...          0    0   
462                                     2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
452                               0            0                       0    0   
462                               0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
452                       0      0                  0               0  
462                       0      0                  0               0  

[2 rows x 55 columns]

In [18]:
df[df['ID'] == '235427']

ID      D.O.A      D.O.D  AGE GENDER RURAL  \
ADMISION                                                   
380       235427  4/26/2017  4/28/2017   56      M     R   
383       235427  4/26/2017  4/28/2017   56      M     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
380                                    O     Apr-17                 3   
383                                    O     Apr-17                 3   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
380                                     1  ...          0    0   
383                                     0  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
380                               0            0                       0    0   
383                               0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
380                       0      0                  0               0  
383                       0      0                  0               0  

[2 rows x 55 columns]

In [19]:
df[df['ID'] == '250680']

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
303       250680   4/24/2017   4/27/2017   85      F     R   
321       250680   4/24/2017   4/27/2017   85      F     U   
6345      250680  01/02/2018  04/02/2018   85      F     R   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
303                                    E     Apr-17                 4   
321                                    E     Apr-17                 4   
6345                                   E     Feb-18                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
303                                     3  ...          0    0   
321                                     0  ...          0    0   
6345                                    1  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
303                               0            0                       0    0   
321                               0            0                       0    0   
6345                              0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
303                       0      0                  0               0  
321                       0      0                  0               0  
6345                      0      0                  0               1  

[3 rows x 55 columns]

In [20]:
# Eliminar filas duplicadas basándose en las columnas 'ID' y 'D.O.A'
df_sin_duplicados = df.drop_duplicates(subset=['ID', 'D.O.A'], keep='first')

In [21]:
#Nuevo dataframe
df_sin_duplicados

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
1         234735    4/1/2017    4/3/2017   81      M     R   
2         234696    4/1/2017    4/5/2017   65      M     R   
3         234882    4/1/2017    4/3/2017   53      M     U   
4         234635    4/1/2017    4/8/2017   67      F     U   
5         234486    4/1/2017   4/23/2017   60      F     U   
...          ...         ...         ...  ...    ...   ...   
15752     469963  31/03/2019  08/04/2019   60      F     U   
15753     699585  31/03/2019  04/04/2019   86      F     U   
15754     699500   3/31/2019    4/1/2019   50      M     R   
15755     700415  31/03/2019  09/04/2019   82      M     U   
15756     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
1                                      E     Apr-17                 3   
2                                      E     Apr-17                 5   
3                                      E     Apr-17                 3   
4                                      E     Apr-17                 8   
5                                      E     Apr-17                23   
...                                  ...        ...               ...   
15752                                  E     Mar-19                 9   
15753                                  O     Mar-19                 5   
15754                                  E     Mar-19                 2   
15755                                  E     Mar-19                10   
15756                                  O     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
1                                       2  ...          0    0   
2                                       2  ...          0    0   
3                                       3  ...          0    0   
4                                       6  ...          0    0   
5                                       9  ...          0    0   
...                                   ...  ...        ...  ...   
15752                                   9  ...          0    0   
15753                                   1  ...          0    0   
15754                                   2  ...          0    0   
15755                                   5  ...          0    0   
15756                                   2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
1                                 0            0                       0    0   
2                                 0            0                       0    0   
3                                 0            0                       0    0   
4                                 0            0                       0    0   
5                                 0            0                       0    0   
...                             ...          ...                     ...  ...   
15752                             0            0                       0    0   
15753                             0            0                       0    0   
15754                             0            0                       0    0   
15755                             0            0                       0    0   
15756                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
1                         0      0                  0               0  
2                         0      0                  0               0  
3                         0      0                  0   

In [22]:
#Verificando si eliminó duplicados
df_sin_duplicados[df_sin_duplicados['ID'] == '250680']

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
303       250680   4/24/2017   4/27/2017   85      F     R   
6345      250680  01/02/2018  04/02/2018   85      F     R   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
303                                    E     Apr-17                 4   
6345                                   E     Feb-18                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
303                                     3  ...          0    0   
6345                                    1  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
303                               0            0                       0    0   
6345                              0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
303                       0      0                  0               0  
6345                      0      0                  0               1  

[2 rows x 55 columns]

In [23]:
df_sin_duplicados[df['ID'] == '235427']

/tmp/ipykernel_21434/4073659580.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_sin_duplicados[df['ID'] == '235427']


ID      D.O.A      D.O.D  AGE GENDER RURAL  \
ADMISION                                                   
380       235427  4/26/2017  4/28/2017   56      M     R   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
380                                    O     Apr-17                 3   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
380                                     1  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
380                               0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
380                       0      0                  0               0  

[1 rows x 55 columns]

In [24]:
#Renombrando dataframe a df1
df1 = df_sin_duplicados.copy()

In [25]:
df1

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
1         234735    4/1/2017    4/3/2017   81      M     R   
2         234696    4/1/2017    4/5/2017   65      M     R   
3         234882    4/1/2017    4/3/2017   53      M     U   
4         234635    4/1/2017    4/8/2017   67      F     U   
5         234486    4/1/2017   4/23/2017   60      F     U   
...          ...         ...         ...  ...    ...   ...   
15752     469963  31/03/2019  08/04/2019   60      F     U   
15753     699585  31/03/2019  04/04/2019   86      F     U   
15754     699500   3/31/2019    4/1/2019   50      M     R   
15755     700415  31/03/2019  09/04/2019   82      M     U   
15756     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
1                                      E     Apr-17                 3   
2                                      E     Apr-17                 5   
3                                      E     Apr-17                 3   
4                                      E     Apr-17                 8   
5                                      E     Apr-17                23   
...                                  ...        ...               ...   
15752                                  E     Mar-19                 9   
15753                                  O     Mar-19                 5   
15754                                  E     Mar-19                 2   
15755                                  E     Mar-19                10   
15756                                  O     Mar-19                 4   

          duration of intensive unit stay  ... CONGENITAL  UTI  \
ADMISION                                   ...                   
1                                       2  ...          0    0   
2                                       2  ...          0    0   
3                                       3  ...          0    0   
4                                       6  ...          0    0   
5                                       9  ...          0    0   
...                                   ...  ...        ...  ...   
15752                                   9  ...          0    0   
15753                                   1  ...          0    0   
15754                                   2  ...          0    0   
15755                                   5  ...          0    0   
15756                                   2  ...          0    0   

          NEURO CARDIOGENIC SYNCOPE  ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  \
ADMISION                                                                        
1                                 0            0                       0    0   
2                                 0            0                       0    0   
3                                 0            0                       0    0   
4                                 0            0                       0    0   
5                                 0            0                       0    0   
...                             ...          ...                     ...  ...   
15752                             0            0                       0    0   
15753                             0            0                       0    0   
15754                             0            0                       0    0   
15755                             0            0                       0    0   
15756                             0            0                       0    0   

          CARDIOGENIC SHOCK  SHOCK PULMONARY EMBOLISM CHEST INFECTION  
ADMISION                                                               
1                         0      0                  0               0  
2                         0      0                  0               0  
3                         0      0                  0   

Finalmente eliminando datos de df(15757 filas) que tenía ID y DOA duplicados en común, se crea un nuevo data frame, df1 (14572 filas), eliminando exactamente 1185 filas duplicadas.

In [26]:
#Calcular la cantidad de ingreso de una persona
df1.sort_index
df1['Nro de Ingresos'] = df1.groupby('ID').cumcount() + 1

In [27]:
#Revisar las personas que ingresaron mas de una vez
df1.sort_values(by='ID')
df1[df1['Nro de Ingresos']>=2]

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
17        413903  01/04/2017  03/04/2017   54      M     U   
91        382159    4/4/2017    4/9/2017   47      F     U   
96         48098    4/5/2017   4/10/2017   65      F     R   
131       384232  08/04/2017  12/04/2017   80      F     U   
148       203622  10/04/2017  17/04/2017   58      F     U   
...          ...         ...         ...  ...    ...   ...   
15729     666139  29/03/2019  02/04/2019   63      F     U   
15739     453768  30/03/2019  10/04/2019   81      M     U   
15740     469288   3/30/2019    4/9/2019   49      M     U   
15749     273016  30/03/2019  31/03/2019   74      F     U   
15752     469963  31/03/2019  08/04/2019   60      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
17                                     O     Apr-17                 3   
91                                     E     Apr-17                 6   
96                                     O     Apr-17                 6   
131                                    E     Apr-17                 5   
148                                    O     Apr-17                 8   
...                                  ...        ...               ...   
15729                                  E     Mar-19                 5   
15739                                  E     Mar-19                12   
15740                                  O     Mar-19                11   
15749                                  E     Mar-19                 2   
15752                                  E     Mar-19                 9   

          duration of intensive unit stay  ... UTI  NEURO CARDIOGENIC SYNCOPE  \
ADMISION                                   ...                                  
17                                      1  ...   0                          0   
91                                      1  ...   0                          0   
96                                      4  ...   0                          0   
131                                     5  ...   0                          0   
148                                     0  ...   1                          0   
...                                   ...  ...  ..                        ...   
15729                                   3  ...   0                          0   
15739                                   6  ...   0                          0   
15740                                   3  ...   0                          0   
15749                                   2  ...   0                          0   
15752                                   9  ...   0                          0   

          ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  CARDIOGENIC SHOCK  SHOCK  \
ADMISION                                                                       
17                  0                       0    0                  0      0   
91                  0                       0    0                  0      0   
96                  0                       0    0                  1      1   
131                 0                       0    0                  0      0   
148                 0                       0    0                  0      0   
...               ...                     ...  ...                ...    ...   
15729               0                       0    0                  0      0   
15739               0                       0    0                  0      0   
15740               0                       0    0                  0      0   
15749               0                       0    0                  0      0   
15752               0                       0    0                  0      0   

          PULMONARY EMBOLISM CHEST INFECTION Nro de Ingresos  
ADMISION                                                      
17                         0        

In [28]:
import numpy as np

In [29]:
#Se crea columna si ingresó o no
df1['REINGRESO'] = np.where(df1['Nro de Ingresos'] > 1, 'Si', 'No')
df1

ID       D.O.A       D.O.D  AGE GENDER RURAL  \
ADMISION                                                     
1         234735    4/1/2017    4/3/2017   81      M     R   
2         234696    4/1/2017    4/5/2017   65      M     R   
3         234882    4/1/2017    4/3/2017   53      M     U   
4         234635    4/1/2017    4/8/2017   67      F     U   
5         234486    4/1/2017   4/23/2017   60      F     U   
...          ...         ...         ...  ...    ...   ...   
15752     469963  31/03/2019  08/04/2019   60      F     U   
15753     699585  31/03/2019  04/04/2019   86      F     U   
15754     699500   3/31/2019    4/1/2019   50      M     R   
15755     700415  31/03/2019  09/04/2019   82      M     U   
15756     699524  31/03/2019  03/04/2019   59      F     U   

         TYPE OF ADMISSION-EMERGENCY/OPD month year  DURATION OF STAY  \
ADMISION                                                                
1                                      E     Apr-17                 3   
2                                      E     Apr-17                 5   
3                                      E     Apr-17                 3   
4                                      E     Apr-17                 8   
5                                      E     Apr-17                23   
...                                  ...        ...               ...   
15752                                  E     Mar-19                 9   
15753                                  O     Mar-19                 5   
15754                                  E     Mar-19                 2   
15755                                  E     Mar-19                10   
15756                                  O     Mar-19                 4   

          duration of intensive unit stay  ... NEURO CARDIOGENIC SYNCOPE  \
ADMISION                                   ...                             
1                                       2  ...                         0   
2                                       2  ...                         0   
3                                       3  ...                         0   
4                                       6  ...                         0   
5                                       9  ...                         0   
...                                   ...  ...                       ...   
15752                                   9  ...                         0   
15753                                   1  ...                         0   
15754                                   2  ...                         0   
15755                                   5  ...                         0   
15756                                   2  ...                         0   

          ORTHOSTATIC  INFECTIVE ENDOCARDITIS  DVT  CARDIOGENIC SHOCK  SHOCK  \
ADMISION                                                                       
1                   0                       0    0                  0      0   
2                   0                       0    0                  0      0   
3                   0                       0    0                  0      0   
4                   0                       0    0                  0      0   
5                   0                       0    0                  0      0   
...               ...                     ...  ...                ...    ...   
15752               0                       0    0                  0      0   
15753               0                       0    0                  0      0   
15754               0                       0    0                  0      0   
15755               0                       0    0                  0      0   
15756               0                       0    0                  0      0   

          PULMONARY EMBOLISM  CHEST INFECTION Nro de Ingresos REINGRESO  
ADMISION                                                                 
1                          0                0               1        No  
2    

ANALISIS DE VARIABLES

In [30]:
import matplotlib.pyplot as plt
import seaborn as sns

In [31]:
#Buscar variables categoricas
categoricas = df1.select_dtypes(include=['object']).columns.tolist()
categoricas

['ID',
 'D.O.A',
 'D.O.D',
 'GENDER',
 'RURAL',
 'TYPE OF ADMISSION-EMERGENCY/OPD',
 'month year',
 'OUTCOME',
 'HB',
 'TLC',
 'PLATELETS',
 'GLUCOSE',
 'UREA',
 'CREATININE',
 'BNP',
 'EF',
 'CHEST INFECTION',
 'REINGRESO']

In [32]:
categoricas_seleccionadas = ['GENDER', 'RURAL', 'TYPE OF ADMISSION-EMERGENCY/OPD', 'OUTCOME', 'CHEST INFECTION', 'REINGRESO']

: 

In [ ]:
# Configuraciones comunes
figsize = (10, 6)
colors = ['skyblue', 'lightgreen']
font_size_title = 14
font_size_labels = 12

# Graficar directamente
ax = df1.plot(
    kind='bar', 
    figsize=figsize, 
    color=colors, 
    title='Distribución de Género y Total de Ingresos por Género', 
    legend=True
)

# Personalizar etiquetas
ax.set_xlabel('Género', fontsize=font_size_labels)
ax.set_ylabel('Reingreso', fontsize=font_size_labels)

# Personalizar el eje X y el grid
ax.tick_params(axis='x', rotation=0)
ax.grid(axis='y', linestyle='--', alpha=0.7)

# Ajustar diseño y mostrar
plt.tight_layout()
plt.show()


In [ ]:
# Graficar directamente sin plt.figure
#ax = df1.plot(kind='bar', figsize=(10, 6), color=['skyblue', 'lightgreen'])

# Personalizar el gráfico
#plt.title('Distribución de Género y Total de Ingresos por Género', fontsize=14)
#plt.xlabel('GENRER', fontsize=12)
#plt.ylabel('REINGRESO', fontsize=12)
#plt.xticks(rotation=0)
#plt.grid(axis='y', linestyle='--', alpha=0.7)
#plt.legend(title='Categoría', loc='upper right')

# Ajustar diseño y mostrar
#plt.tight_layout()
#plt.show()


Exception ignored in: <bound method IPythonKernel._clean_thread_parent_frames of <ipykernel.ipkernel.IPythonKernel object at 0x762c8e2bfdd0>>
Traceback (most recent call last):
  File "/home/codespace/.local/lib/python3.12/site-packages/ipykernel/ipkernel.py", line 775, in _clean_thread_parent_frames
    def _clean_thread_parent_frames(

KeyboardInterrupt: 


In [33]:
numericas = df1.select_dtypes(include=['int']).columns.tolist()
numericas

['AGE',
 'DURATION OF STAY',
 'duration of intensive unit stay',
 'SMOKING ',
 'ALCOHOL',
 'DM',
 'HTN',
 'CAD',
 'PRIOR CMP',
 'CKD',
 'RAISED CARDIAC ENZYMES',
 'SEVERE ANAEMIA',
 'ANAEMIA',
 'STABLE ANGINA',
 'ACS',
 'STEMI',
 'ATYPICAL CHEST PAIN',
 'HEART FAILURE',
 'HFREF',
 'HFNEF',
 'VALVULAR',
 'CHB',
 'SSS',
 'AKI',
 'CVA INFRACT',
 'CVA BLEED',
 'AF',
 'VT',
 'PSVT',
 'CONGENITAL',
 'UTI',
 'NEURO CARDIOGENIC SYNCOPE',
 'ORTHOSTATIC',
 'INFECTIVE ENDOCARDITIS',
 'DVT',
 'CARDIOGENIC SHOCK',
 'SHOCK',
 'PULMONARY EMBOLISM',
 'Nro de Ingresos']

In [32]:
df_correlaciones = df1.copy()

In [ ]:
df1.isnull().sum()

ID                                    0
D.O.A                                 0
D.O.D                                 0
AGE                                   0
GENDER                                0
RURAL                                 0
TYPE OF ADMISSION-EMERGENCY/OPD       0
month year                            0
DURATION OF STAY                      0
duration of intensive unit stay       0
OUTCOME                               0
SMOKING                               0
ALCOHOL                               0
DM                                    0
HTN                                   0
CAD                                   0
PRIOR CMP                             0
CKD                                   0
HB                                  245
TLC                                 276
PLATELETS                           277
GLUCOSE                             837
UREA                                236
CREATININE                          241
BNP                                7912
